compiler/rustc_smir/src/rustc_internal/mod.rs

```diff
@@ -10,7 +10,7 @@ use crate::{
 use rustc_middle::ty::TyCtxt;
 pub use rustc_span::def_id::{CrateNum, DefId};
 
-fn with_tables<R>(mut f: impl FnMut(&mut Tables<'_>) -> R) -> R {
+pub(crate) fn with_tables<R>(mut f: impl FnMut(&mut Tables<'_>) -> R) -> R {
```

Comment:
I wanted to keep this private for now, to have a stronger separation. Not sure yet whether that is good long term, but maybe let's keep it for now?

Reply:
Ok, do you have any suggestion on how to get the context when translating from an internal rustc type that requires context information?

From what I can tell, we need to modify the Tables when translation from MIR -> SMIR and we need to read the Tables when translating SMIR -> MIR. So I was thinking that this function should be available for the `rustc_smir` and the `rustc_internal` to do their respective translations.

Reply:
@oli-obk?

Reply:
The wrapper function you removed was my attempt at keeping everything but the wrappers in that module. As long as that isn't too annoying, I think we should keep adding such wrapper functions as needed. We can always remove all of them and expose the tables function directly, but that should be its own PR with its own motivation.

Reply:
That makes sense. Let me revert that. Thanks